### PR TITLE
Return a Daru::Vector if multiple values present in mode.

### PR DESCRIPTION
**Pull request by dylandy**
Reference to  issue #272. Not sure if I wrote it correctly. :P

### CODE REVIEW

**Comment by v0dro** [comment]
Can you make Travis pass?

File: lib/daru/maths/statistics/vector.rb [hunk 1]
```diff
@@ -30,7 +30,9 @@ def median
         end
 
         def mode
-          frequencies.max { |a,b| a[1]<=>b[1] }.first
+          max = frequencies.values.max
```
Comment (v0dro) [inline lib/daru/maths/statistics/vector.rb]: Don't call `values` use `frequencies.max { |a,b| a[1] <=> b[1] }.last`

File: lib/daru/maths/statistics/vector.rb [hunk 2]
```diff
@@ -30,7 +30,9 @@ def median
...
+          output = frequencies.select { |_k,v| v == max }.to_a.map(&:first)
```
Comment (v0dro) [inline lib/daru/maths/statistics/vector.rb]: You're calling `frequencies` twice. That's expensive. Cache it .

Comment (v0dro) [inline lib/daru/maths/statistics/vector.rb]: No need to call `to_a.map`. Just `.values` will suffice since `select` will return a `Hash`.

**Comment by dylandy** [comment]
Hello, I have passed the Travis test after the commit this time.

**Comment by dylandy** [comment]
I have no idea why I failed on CI test after merging this time.

**Comment by lokeshh** [comment]
See https://github.com/v0dro/daru/pull/279#issuecomment-258661492

**Comment by v0dro** [comment]
I think if you rebase from the master branch your tests should pass now. Can you please do that?

File: lib/daru/maths/statistics/vector.rb [hunk 3]
```diff
@@ -32,7 +32,10 @@ def median
         end
 
         def mode
-          frequencies.to_h.max { |a,b| a[1]<=>b[1] }.first
+          freq = frequencies.to_h
+          max = freq.max { |a,b| a[1] <=> b[1] }.last
```
Comment (zverok) [inline lib/daru/maths/statistics/vector.rb]: Use `max_by`, please.

**Comment by v0dro** [comment]
The build is still failing. Please make it pass.

**Comment by dylandy** [comment]
Sorry for disappear too much time. I was too busy to do open source project recently. 

I just came up with following problem when I tried  ```rake spec```. I have no idea to fix this problem.

**Comment by zverok** [comment]
@dylandy GSLWrapper is not defined if you have no GSL library installed: https://github.com/SciRuby/daru/blob/master/lib/daru/accessors/gsl_wrapper.rb#L1

I believe you need to install GSL to be fully able to take a part in development. Though, as a temporary solution, you can add the same `if Daru.has_gsl?` at the top of spec/accessors/gsl_wrapper_spec.rb

**Comment by dylandy** [comment]
well, I finally fixed my library issue and it seems like somebody just solved this pr.
Thanks a lot.

**Pull request closed without merging**